### PR TITLE
[ENGG-2664] JSON body getting copied in RAW editor - API Client

### PR DESCRIPTION
**Pull request by Aarushsr12**
- Json body will not be copied in RAW body
- Also when switching content type the prettification will be applied

### CODE REVIEW

File: app/src/features/apiClient/screens/apiClient/components/clientView/components/request/RequestBody.tsx [hunk 1]
```diff
@@ -37,6 +38,13 @@ const RequestBody: React.FC<Props> = ({ body, contentType, variables, setRequest
     setRequestEntry((prev) => ({ ...prev, request: { ...prev.request, body: updatedBody } }));
   }, [contentType, rawBody, jsonBody, formBody, setRequestEntry]);
 
+  useEffect(() => {
+    if (contentType === RequestContentType.JSON && typeof jsonBody === "string") {
+      const prettifiedContent = prettifyCode(jsonBody, EditorLanguage.JSON).code;
+      setJsonBody(prettifiedContent);
+    }
+  }, [contentType]);
+
```
Comment (RuntimeTerror10) [inline app/src/features/apiClient/screens/apiClient/components/clientView/components/request/RequestBody.tsx:47]: Why is this required?

Comment (Aarushsr12) [reply app/src/features/apiClient/screens/apiClient/components/clientView/components/request/RequestBody.tsx:47]: Issue: when switching between RAW & json, the prettification gets removed
Solution: added a useEffect on json content type, which will prettify when switching the content type to JSON

**Review by RuntimeTerror10 — APPROVED** [review]
